sacc2/sacc.py

```diff
@@ -0,0 +1,940 @@
+        """
+        Add a pre-constructed BaseTracer instance to this data set.
+        If you just have, for example the z and n(z) data then
+        use the add_tracer method instead.
+
+        Parameters
+        ----------
+
+        tracer: Tracer instance
+            The tracer object to add to the data set
+        """
+        self.tracers[tracer.name] = tracer
+
+    def add_data_point(self, data_type, tracers, value, tracers_later=False, **tags):
+        """
+        Add a data point to the set.
+
+        Parameters
+        ----------
+        data_type: str
+
+        tracers: tuple of str
+            Strings corresponding to zero or more of the tracers in the data set
+            These should either be already set up using the add_tracer method,
+            or you could set tracers_laster=True if you want to add them later.
+            e.g. for 2pt measurements the tracers are the names of the two n(z)
+            samples
+
+        value: float
+            A single value for the data point
+
+        tracers_later: bool
+            If True, do not complain if the tracers are not know already
+
+        **tags:
+            Tags to apply to this data point.
+            Tags can be any arbitrary metadata that you might want later,
+            For 2pt data the tag would include an angle theta or ell.
+
+        Returns
+        -------
+        None
+        """
+        if self.covariance is not None:
+            raise ValueError("You cannot add a data point after setting the covariance")
+        tracers = tuple(tracers)
+        for tracer in tracers:
+            if (tracer not in self.tracers) and (not tracers_later):
+                raise ValueError(f"Tracer named '{tracer}' is not in the known list of tracers."
+                                 "Either put it in before adding data points or set tracers_later=True")
+        d = DataPoint(data_type, tracers, value, **tags)
+        self.data.append(d)
+
+    def add_covariance(self, covariance):
+        """
+        Once you have finished adding data points, add a covariance
+        for the entire set.
+
+        Parameters
+        ----------
+        covariance: array or list
+            2x2 numpy array containing the covariance of the added data points
+            OR a list of blocks
+
+        Returns
+        -------
+        None
+        """
+        if isinstance(covariance, BaseCovariance):
+            self.covariance = covariance
+        else:
+            self.covariance = BaseCovariance.make(covariance, len(self))
+
+    def _indices_to_bool(self, mask):
+        # Convert an array of indices into a boolean True mask
+        if mask.dtype not in [np.int8, np.int16, np.int32, np.int64]:
+            raise ValueError(f"Wrong mask type ({mask.  dtype}) - expected integers or boolean")
+        m = np.zeros(len(self), dtype=bool)
+        for i in mask:
+            m[i] = True
+        return m
+
+    def mask_indices(self, indices):
```

Comment:
So this function masks where the input mask is False? That def seems backwards. Maybe name it `select_indices` or something?

Reply:
Blame @slosar and @damonge ! How about `keep_indices` and `remove_indices`?

Reply:
Yup. keep and remove are great.

Reply:
TBH, we only need one.

Reply:
Changed to keep and remove.  It's no extra maintenance cost to support both since one just calls the other.